sdks/python/apache_beam/runners/portability/job_server.py

```diff
@@ -141,18 +147,21 @@ def path_to_beam_jar(gradle_target, artifact_id=None):
         gradle_target, artifact_id=artifact_id)
 
   @staticmethod
-  def local_jar(url):
-    return subprocess_server.JavaJarServer.local_jar(url)
+  def local_jar(url, jar_cache_dir=None):
+    return subprocess_server.JavaJarServer.local_jar(url, jar_cache_dir)
 
   def subprocess_cmd_and_endpoint(self):
-    jar_path = self.local_jar(self.path_to_jar())
+    jar_path = self.local_jar(self.path_to_jar(), self._jar_cache_dir)
     artifacts_dir = (
         self._artifacts_dir if self._artifacts_dir else self.local_temp_dir(
             prefix='artifacts'))
     job_port, = subprocess_server.pick_port(self._job_port)
-    subprocess_cmd = [self._java_launcher, '-jar'] + self._jvm_properties + [
-        jar_path
-    ] + list(
-        self.java_arguments(
-            job_port, self._artifact_port, self._expansion_port, artifacts_dir))
+    subprocess_cmd = [self._java_launcher, '-jar'
+                      ] + self._jvm_properties + [jar_path] + list(
+                          self.java_arguments(
+                              job_port,
+                              self._artifact_port,
+                              self._expansion_port,
+                              artifacts_dir,
+                              self.jar_cache_dir))
```

Comment:
```suggestion
                              self._jar_cache_dir))
```

Reply:
Looks like this is causing precommit failures

Reply:
Oh, I'm sorry for that. Thank you 👍